dijkstra/lib/graph.js

```diff
@@ -15,20 +29,16 @@ function generate(numNodes) {
   for (var i=0; i < numNodes; i++) {
     var node = [];
 
-    // assure every node is connected to at least one other by making a circle
+    // make circular edges
+    //  - assures every node is reachable from every other node
+    //  - assures every node has at least one incoming and one outgoing edge
     node.push(makeEdge(i === numNodes - 1 ? 0 : i + 1));
 
-    for (var e=0; e < getRandomInt(0, 4); e++) {
+    // make random edges
+    const edges = getRandomInt(0, MAX_RANDOM_EDGES);
```

Comment:
If you're going with the ES6 syntax like that, the best practice is to ban the word `var` completely and use either `const` or `let`. `let` has the advantage of being block-scoped (so it doesn't leak outside of loops etc.)

Reply:
Nit: `edges` is not a good variable name because it implies an array of edge objects rather than its number. I'd go for something like `numEdges`.

Reply:
Would you advocate for using `for (let i = 0 ...` as well instead of `for var (i = 0 ...`?

Reply:
Yes. There's even a rule for that in ESLint: https://eslint.org/docs/rules/no-var